testing-unit-fs-mocks/Testing.fs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+namespace UnitTesting
+
+open System
+open System.Collections.Generic
+open System.Collections.Immutable
+open System.Threading.Tasks
+open Pulumi.Testing
+
+type Mocks() =
+    interface IMocks with
+        member this.NewResourceAsync(typeName: string, name: string, inputs: ImmutableDictionary<string, obj>, provider: string, id: string): Task<ValueTuple<string,obj>> =
+            let inputKVs = inputs |> Seq.map(fun kv -> kv.Key, kv.Value) |> Seq.toList
+            let nameKVs = if inputs.ContainsKey("name") then [] else [("name", name :> obj)]
+            let endpointKVs =
+                if typeName = "azure:storage/account:Account"
```

Comment:
Do we not need the same removal for source from the Blob as per the CSharp example?

Reply:
It happens (in slightly weird way to avoid mutations) on line 24

Reply:
ah... ok :) That's an Fsharp-ism that I didn't get :)